File: matlab/src/DEFAULTS.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% matlab using latex
set(groot,'defaulttextinterpreter','latex');
set(groot, 'defaultAxesTickLabelInterpreter','latex');
set(0,'defaulttextInterpreter','latex')
set(groot, 'defaultLegendInterpreter','latex');
set(groot,'DefaultAxesTitle','latex');
% This script changes all interpreters from tex to latex. 
list_factory = fieldnames(get(groot,'factory'));
index_interpreter = find(contains(list_factory,'Interpreter'));
for i = 1:length(index_interpreter)
    default_name = strrep(list_factory{index_interpreter(i)},'factory','default');
    set(groot, default_name,'latex');
end
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% load library and parameters
%addpath(genpath('../../lib/'),genpath('../../params/'))
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% directories
pic_dir="./images/";
display("[Defaults] images will be stored in: "+ pic_dir)

dat_dir="./data/";
display("[Defaults] images will be stored in: "+ dat_dir)


% Set default axes properties
set(0, 'DefaultAxesFontSize', 14);        % Larger font size for axes
set(0, 'DefaultAxesLineWidth', 1.5);      % Thicker axes lines
set(0, 'DefaultAxesXMinorTick', 'on');    % Enable minor ticks
set(0, 'DefaultAxesYMinorTick', 'on');

% Set default line properties for plots
set(0, 'DefaultLineLineWidth', 2);        % Thicker plot lines
set(0, 'DefaultLineMarkerSize', 8);       % Larger markers

% Set default legend properties
set(0, 'DefaultLegendFontSize', 14);      % Larger font for legends

File: python/src/DEFAULTS.py
### Matplotlib plot parameters ###

import matplotlib.pyplot as plt

plt.rcParams.update({
    "text.usetex": True,                # Use LaTeX for all text
    "font.size": 14,                    # Default font size
    "axes.linewidth": 1.5,              # Thicker axes lines
    "xtick.minor.visible": True,        # Enable minor ticks
    "ytick.minor.visible": True,
    "lines.linewidth": 2,               # Thicker plot lines
    "lines.markersize": 8,              # Larger markers
    "legend.fontsize": 14               # Larger font for legends
})

### Data & images directories definition ###

import os

pic_dir = "./images/"
dat_dir = "./data/"

print(f"[Defaults] Images will be stored in: {pic_dir}")
print(f"[Defaults] Data will be stored in: {dat_dir}")

# Create directories if they don’t exist
os.makedirs(pic_dir, exist_ok=True)
os.makedirs(dat_dir, exist_ok=True)